src/apps/config/leader.lua

```diff
@@ -169,8 +169,9 @@ local function path_setter_for_grammar(grammar, path)
       assert(grammar.type == 'struct')
       local tail_id = data.normalize_id(tail_name)
       return function(config, subconfig)
-         getter(config)[tail_id] = subconfig
-         return config
+	 local newconfig = lib.deepcopy(config)
```

Comment:
Right now the expectation is that there is a function that goes from the YANG configuration to the app graph, and that function transfers any information relevant to the data plane from the YANG configuration to the app graph.  Here what I hear you saying is that there is now a bit of information that's present in the YANG configuration (the instance name) but not present in the app graph -- is that right?

As far as I understand things, with the current data plane if someone sets the 'name' property, there's no need to restart any particular app -- they don't care what name the instance is bound to.  So the system appears (AFAIU) to be operating correctly there.  Do you agree?

To get the previous name it shouldn't be necessary to have the previous configuration in any case; since it's a global property, we can just store that data in a global variable in core/app.lua.  WDYT?

AFAIU also the follower doesn't need the name.  Is that correct?

Then I think the setup function should effectively do app.claim_name() on the name in the incoming yang configuration.  That makes the setup function have a side effect, but oh well.  WDYT?

Apologies for the long list of ignorant questions on my part :)